diff --git a/cvl/internal/yparser/yparser.go b/cvl/internal/yparser/yparser.go
index 92343bc..6bb34f3 100644
--- a/cvl/internal/yparser/yparser.go
+++ b/cvl/internal/yparser/yparser.go
@@ -38,7 +38,6 @@ import (
 #include <stdio.h>
 #include <string.h>

-extern int lyd_check_mandatory_tree(struct lyd_node *root, struct ly_ctx *ctx, const struct lys_module **modules, int mod_count, int options);

 struct lyd_node* lyd_parse_data_path(struct ly_ctx *ctx,  const char *path, LYD_FORMAT format, int options) {
        return lyd_parse_path(ctx, path, format, options);
@@ -51,16 +50,6 @@ struct lyd_node *lyd_parse_data_mem(struct ly_ctx *ctx, const char *data, LYD_FO

 int lyd_data_validate(struct lyd_node **node, int options, struct ly_ctx *ctx)
 {
-       int ret = -1;
-
-       //Check mandatory elements as it is skipped for LYD_OPT_EDIT
-       ret = lyd_check_mandatory_tree(*node, ctx, NULL, 0, LYD_OPT_CONFIG | LYD_OPT_NOEXTDEPS);
-
-       if (ret != 0)
-       {
-               return ret;
-       }
-
        return lyd_validate(node, options, ctx);
 }

